### ISSUE TITLE
consults in Imfusion

### ISSUE THREAD

**Issue by lkccu** [mfazampour/medphys_ct_us_registration#1]
Hi author @mfazampour ,thanks for your great work in spine us-ct generation which is a great tutorial and pioneer in this area in navigation system. I am trying to run your code right now.

However, I found this code executed in Imfusion. I am wondering which version and products you bought from imfusion. It will be greatly helpful if you inform me that 'cause it is so expensive.

Thank you.

**Comment by mfazampour** [comment]
Hi, Thanks for reaching out. We use the academic installer. If you are a researcher at a university, your PI can contact them and ask them about the pricing. I hope they can offer you a good price.

**Comment by lkccu** [comment]
Hi @mfazampour, there is another problem I encountered.  When reruning the part _"DeformSegmentationAndSimulate"_ in program, the tutorial asked me to generate segs using TotalSegmentator. 

However I am confused which version to use and is there any additional operation to execute other than segmenting them by `--ta total` and `totalseg_combine_masks -i totalsegmentator_output_dir -o combined_mask.nii.gz `  to generate a combined segmentation labelmaps?

Also I found some differences between the list Doc.Wasserth opened in [TotalSegmentator class-details](https://github.com/wasserth/TotalSegmentator#class-details) and your code. In your code:
 `bone_data = torch.where
(segmentation_data == 92 | ((segmentation_data >= 18) & (segmentation_data < 27)), 1.0, 0.0)` indicates label:18 in labelmap is part of bone material. However in list label:18 denotes small_bowel in list. So it is difficult for me to make out.

Thank you for your great work once again. If you would give me a hand, I will be so much appreciated.

regards,
jack lin.

**Comment by lkccu** [comment]
p.s. I found nowhere in repo [TotalSegmentator](https://github.com/wasserth/TotalSegmentator) to help me generate a .nrrd medi. data by TotalSegmentator. Why take .nrrd file for input format instead of .nii.gz format?

**Comment by mfazampour** [comment]
Hi, I used the slicer plugin of total segmentator, and it was generating nrrd images by default. It might also be that the numbering of the labels is different a bit there.

**Comment by lkccu** [comment]
Hi,thank you for your reply. I have solved this problem by editing the format of infile.

However, I have encountered an another problem unfortunately this days. It is caused somehow by **_Imfusion Suite ultrasound_** software. I have reruned _04_ultrasound_simulation.py_ in folder _DeformSegmentationAndSimulate_ . A set of 2d-ultrasound images was expected to be simulated from **seg_sim.nii.gz, however, only one slice of ct segmentation has been simulated which seems like the last slice of segmentation. I dont know what I should do to deal with it. Could u give me a hand?

regards.
